pkg/cloud/logs/logs.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logs
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/go-logr/logr"
+)
+
+const (
+	logWithHTTPHeader = 9
+	logWithHTTPBody   = 10
+)
+
+func GetAWSLogLevel(logger logr.Logger) aws.LogLevelType {
+	if logger.V(logWithHTTPBody).Enabled() {
+		return aws.LogDebugWithHTTPBody
+	}
+
+	if logger.V(logWithHTTPHeader).Enabled() {
+		return aws.LogDebug
+	}
+
+	return aws.LogOff
+}
+
+func NewWrapLogr(logger logr.Logger) aws.Logger {
+	return &logrWrapper{
+		log: logger,
+	}
+}
+
+type logrWrapper struct {
+	log logr.Logger
+}
+
+func (l *logrWrapper) Log(msgs ...interface{}) {
```

Comment:
Can this be used here/can this be refactored/is this relevant:

https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/38007acdd23ed4cc091cb1f267b178fd195bd8ae/pkg/cloud/services/ec2/instances.go#L609-L616

Reply:
Can be a separate PR

Reply:
Yes i think we can refactor this....lets do it as a follow up PR